scripts/run-azurite.ps1

```diff
@@ -0,0 +1,100 @@
+<#
+NOTE: This does take measurable time to run, evaluate whether running
+repeatedly in CI is preferable to just having an image that has had it run
+to complete installation with it remaining available in that image (runtime
+cost when CI instances currently overloaded versus required image storage -
+and whether other dependencies might also require nodejs separately from
+azurite.
+
+#>
+
+<#
+.SYNOPSIS
+This is a Powershell script to run previously installed azurite on Windows.
+
+.DESCRIPTION
+This is a Powershell script to run previously installed azurite server on Windows.
+
+.LINK
+https://github.com/TileDB-Inc/TileDB
+#>
+
+[CmdletBinding()]
+Param(
+)
+
+# Return the directory containing this script file.
+function Get-ScriptsDirectory {
+    Split-Path -Parent $PSCommandPath
+}
+
+$TileDBRootDirectory = Split-Path -Parent (Get-ScriptsDirectory)
+$InstallPrefix = Join-Path $TileDBRootDirectory "dist"
+$StagingDirectory = Join-Path (Get-ScriptsDirectory) "deps-staging"
+
+function DownloadFile([string] $URL, [string] $Dest) {
+    Write-Host "Downloading $URL to $Dest..."
+    Invoke-WebRequest -Uri $URL -OutFile $Dest
+}
+
+function DownloadIfNotExists([string] $Path, [string] $URL) {
+    if (!(Test-Path $Path)) {
+        # TLS1.2 seems to work for nodejs site
+        [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+        DownloadFile $URL $Path
+    }
+}
+
+function run-batchfile-modifying-path([string] $scriptPath) {
+    Set-StrictMode -Version 3
+    $tempfile = [IO.Path]::GetTempFileName()
+    cmd /c " `"$scriptPath`" && set > `"$tempfile`" "
+    Get-Content $tempfile | ForEach-Object {
+        if($_ -match "^(PATH)=(.*)$") {
+            Set-Content "env:PATH" $matches[2]
+        }
+    }
+    Remove-Item $tempfile
+}
+
+
+function run-azurite() {
+    $azuriteDataPath= (Join-Path $env:TEMP "azurite-data")
+    $NodeJSGenericInstalledPath = Join-Path $InstallPrefix "NodeJS"
+    if(!(Test-Path $NodeJSGenericInstalledPath)) {
+	    Write-Host "Unable to find $NodeJSGenericInstalledPath!"
+	}
+    #presumes that the generic 'NodeJS' link to the versioned install directory exists.
+    $NodeJSInitBatchFile = (Join-Path $NodeJSGenericInstalledPath "nodevars.bat")
+    Write-Host "NodeJSGenericInstalledPath: " + $NodeJSGenericInstalledPath
+
+    #need this to have access to 'node' of nodejs...
+    run-batchfile-modifying-path $NodeJSInitBatchFile
```

Comment:
Please use `$env:Path += ...`, see https://github.com/TileDB-Inc/TileDB/blob/dev/scripts/azure-windows.yml#L19

Reply:
I don't think that would be appropriate here, please reconsider this request.
The $NodeJSInitBatchFile (part of the node installation for init'ing 'node's environment) is a black box to tiledb.
To use $env:Path would require looking inside that black box and duplicating its logic, 
with that logic potentially changing outside the control and awareness of tiledb.
Using the routine as it is captures whatever that black box may do to the environment path now or in the future.

Reply:
I see, ok... Backing up a bit, it seems like we can simplify this a lot with the azure pipelines`NodeTool` task: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/node-js?view=azure-devops

Reply:
But that tool is only available for the azure pipeline, correct?
Which would not work for a developer trying to use the install-run-azurite.ps1/run-azurite.ps1 locally for development/testing....
ATM, those scripts are currently only optionally left in the azure-windows.yml to be able to mimic what a developer might do in the event some problem should arise due to a difference in developer environment (having used install-run-azurite.ps1) vs azure pipeline environment (currently actively using the vm image pre-installed nodejs version).

Reply:
I have not tried without the path modifications since turning the azurite install into a -g(lobal) install which was necessary to get that codepath to work correctly in azure pipeline, where a non-g (no -g on install) did work for me locally.
I will try that and see if the path modification as written is still necessary since that change.

Reply:
tried - the -g(lobal) install of azurite did not eliminate the need for path to include whatever the 'nodevars.bat' of the installed nodejs sets.
repeating - script now primarily to aid in developer local environment setup.  Assuming the azure pipelines nodejs tool not useful in that scenario.

Reply:
Ok, marking as resolved for now.